Add operator object storage docs (#2687)
grafana/tempo: docs/sources/tempo/setup/operator/object-storage.md

---

> title: 'Object storage'

Reviewer: The single quotes aren't needed around a title in front matter. 
```suggestion
title: Object storage
```

description: Setup for storing traces to Object Storage
menuTitle: Object storage
weight: 200
aliases:
- /docs/tempo/operator/object-storage
---

> 

Reviewer: I'm adding a page title that matches the title set in the front matter. 
```suggestion

# Object storage for Operator
```

### Requirements

* Create a [project](https://cloud.google.com/resource-manager/docs/creating-managing-projects) on Google Cloud Platform.
* Create a [bucket](https://cloud.google.com/storage/docs/creating-buckets) under same project.
* Create a [service account](https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account) under same project for GCP authentication.

### Installation

* Deploy the Tempo Operator to your cluster.

* Copy the service account credentials received from GCP into a file name `key.json`.

* Create an Object Storage secret with keys `bucketname` and `key.json` as follows:

    ```console
    kubectl create secret generic tempostack-dev-gcs \
      --from-literal=bucketname="<BUCKET_NAME>" \
      --from-file=key.json="<PATH/TO/KEY.JSON>"
    ```

  where `tempostack-dev-gcs` is the secret name, `<BUCKET_NAME>` is the name of bucket created in requirements step and `<PATH/TO/KEY.JSON>` is the file path where the `key.json` was copied to.

* Create an instance of TempoStack by referencing the secret name and type as `gcs`:

  ```yaml
  spec:
    storage:
      secret:
        name: tempostack-dev-gcs
        type: gcs
  ```

> ## Minio

Reviewer: According to the MinIO website (linked below), the capitalization should be MinIO. 
```suggestion
## MinIO
```

### Requirements

> * Deploy Minio on your Cluster, e.g. using the [Minio Operator](https://operator.min.io/)
> 
> * Create a [bucket](https://docs.min.io/docs/minio-client-complete-guide.html) on Minio via CLI.

Reviewer: ```suggestion
* Deploy MinIO on your cluster using the [Minio Operator](https://operator.min.io/) or another method. 

* Create a [bucket](https://docs.min.io/docs/minio-client-complete-guide.html) on MinIO via CLI.
```
